src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -108,7 +115,8 @@
    :borrow-timeout schema/Int
    :max-active-instances schema/Int
    :max-requests-per-instance schema/Int
-   :use-legacy-auth-conf schema/Bool})
+   :use-legacy-auth-conf schema/Bool
+   (schema/optional-key :environment-class-cache-enabled) schema/Bool})
```

Comment:
I'd rather dissoc than have an optional key, and I'd rather dissoc than pass it around to places where it doesn't make sense... it seems like the issue here is that we actually need two schemas now, one for what is allowed in the config file and one for what we pass around for things that are actually related to jruby.  That is the pattern that I would prefer that we strive for in the future, personally, but it's probably too much work for this PR since we are on a time crunch.